com.unity.ml-agents/Tests/Editor/ModelRunnerTest.cs

```diff
@@ -0,0 +1,100 @@
+
+        private BrainParameters GetContinuous2vis8vec2actionBrainParameters()
+        {
+            var validBrainParameters = new BrainParameters();
+            validBrainParameters.vectorObservationSize = 8;
+            validBrainParameters.vectorActionSize = new int[] { 2 };
+            validBrainParameters.numStackedVectorObservations = 1;
+            validBrainParameters.vectorActionSpaceType = SpaceType.Continuous;
+            return validBrainParameters;
+        }
+
+        private BrainParameters GetDiscrete1vis0vec_2_3action_recurrModelBrainParameters()
+        {
+            var validBrainParameters = new BrainParameters();
+            validBrainParameters.vectorObservationSize = 0;
+            validBrainParameters.vectorActionSize = new int[] { 2, 3 };
+            validBrainParameters.numStackedVectorObservations = 1;
+            validBrainParameters.vectorActionSpaceType = SpaceType.Discrete;
+            return validBrainParameters;
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            continuous2vis8vec2actionModel = (NNModel)AssetDatabase.LoadAssetAtPath(k_continuous2vis8vec2actionPath, typeof(NNModel));
+            discrete1vis0vec_2_3action_recurrModel = (NNModel)AssetDatabase.LoadAssetAtPath(k_discrete1vis0vec_2_3action_recurrModelPath, typeof(NNModel));
+            var go = new GameObject("SensorA");
+            sensor_21_20_3 = go.AddComponent<Test3DSensorComponent>();
+            sensor_21_20_3.Sensor = new Test3DSensor("SensorA", 21, 20, 3);
+            sensor_20_22_3 = go.AddComponent<Test3DSensorComponent>();
+            sensor_20_22_3.Sensor = new Test3DSensor("SensorA", 20, 22, 3);
+        }
+
+        [Test]
+        public void TestModelExist()
+        {
+            Assert.IsNotNull(continuous2vis8vec2actionModel);
+            Assert.IsNotNull(discrete1vis0vec_2_3action_recurrModel);
+        }
+
+        [Test]
+        public void TestCreation()
+        {
+            var modelRunner = new ModelRunner(continuous2vis8vec2actionModel, GetContinuous2vis8vec2actionBrainParameters());
+            modelRunner.Dispose();
+            modelRunner = new ModelRunner(discrete1vis0vec_2_3action_recurrModel, GetDiscrete1vis0vec_2_3action_recurrModelBrainParameters());
+            modelRunner.Dispose();
+        }
+
+        [Test]
+        public void TestHasModel()
+        {
+            var modelRunner = new ModelRunner(continuous2vis8vec2actionModel, GetContinuous2vis8vec2actionBrainParameters(), InferenceDevice.CPU);
+            Assert.True(modelRunner.HasModel(continuous2vis8vec2actionModel, InferenceDevice.CPU));
+            Assert.False(modelRunner.HasModel(continuous2vis8vec2actionModel, InferenceDevice.GPU));
+            Assert.False(modelRunner.HasModel(discrete1vis0vec_2_3action_recurrModel, InferenceDevice.CPU));
+            modelRunner.Dispose();
+        }
+
+        [Test]
+        public void TestRunModel()
```

Comment:
Worth running the other model too? This will cover at least some of the tensor generator/appliers, right?